src/equal.h

```diff
@@ -4,6 +4,9 @@
 #include "vctrs.h"
 
 
+static inline bool lgl_scalar_equal_missing(int x) {
```

Comment:
Follow the surrounding `_equal_scalar_` naming scheme?

Reply:
Did you add these variants for performance or for readability? I would assume not performance since `NA_LOGICAL` etc are not compile time constants anyway.

Reply:
Mainly for readability, but also:
- I think the double one is more correct this way actually, something like this doesn't cover the `NaN` case? It should also be faster https://github.com/r-lib/vctrs/blob/05968ce8e669f73213e3e894b5f4424af4f46316/src/dictionary.c#L54-L56
- the character variant is faster because it doesn't try to translate
- the list variant is faster because we don't have to use `equal_object()`, and we don't have to pass in and subset `vctrs_shared_na_list`
- it is nice to have a raw variant that just returns false